src/sentry/static/sentry/app/views/organizationEventsV2/transactionView/span.tsx

```diff
@@ -0,0 +1,648 @@
+import React from 'react';
+import styled from 'react-emotion';
+import {get} from 'lodash';
+
+import {t} from 'app/locale';
+import space from 'app/styles/space';
+import Count from 'app/components/count';
+import Tooltip from 'app/components/tooltip';
+
+import {SPAN_ROW_HEIGHT, SpanRow} from './styles';
+import {
+  MINIMAP_CONTAINER_HEIGHT,
+  MINIMAP_SPAN_BAR_HEIGHT,
+  MIN_NUM_OF_SPANS_TO_MOVE_MINIMAP,
+} from './minimap';
+
+import {
+  toPercent,
+  SpanBoundsType,
+  SpanGeneratedBoundsType,
+  getHumanDuration,
+  parseSpanTimestamps,
+  TimestampStatus,
+} from './utils';
+import {SpanType, ParsedTraceType} from './types';
+import {DividerHandlerManagerChildrenProps} from './dividerHandlerManager';
+import SpanDetail from './spanDetail';
+
+type PropType = {
+  span: Readonly<SpanType>;
+  trace: Readonly<ParsedTraceType>;
+  generateBounds: (bounds: SpanBoundsType) => SpanGeneratedBoundsType;
+  treeDepth: number;
+  numOfSpanChildren: number;
+  renderedSpanChildren: Array<JSX.Element>;
+  spanBarColour: string;
+  spanNumber: number;
+
+  dividerHandlerChildrenProps: DividerHandlerManagerChildrenProps;
+};
+
+type State = {
+  displayDetail: boolean;
+  showSpanTree: boolean;
+};
+
+const INTERSECTION_THRESHOLDS: Array<number> = [];
+
+for (let i = 0; i <= 1.0; i += 0.01) {
+  INTERSECTION_THRESHOLDS.push(i);
+}
+
+class Span extends React.Component<PropType, State> {
+  state: State = {
+    displayDetail: false,
```

Comment:
should `showDetail` be a bit clearer / more in line with `showSpanTree`?

Reply:
@lynnagara I don't quite follow; are you suggesting to rename `showDetail` to `showSpanTree`? 🤔 

If so, it would be the wrong name since span-descendants aren't displayed within the span detail component.

Reply:
Haha no sorry, I thought `showDetail` might be better than `displayDetail`, since we use the `show-` prefix with the span tree stuff

Reply:
Done in https://github.com/getsentry/sentry/pull/14225/commits/3dac2a6476a7bff901d636345b975eaf84f13179